Sources/SpeziViews/Views/ManagedNavigationStack/NavigationFlowBuilder.swift

```diff
@@ -0,0 +1,55 @@
+//
+// This source file is part of the Stanford Spezi open-source project
+//
+// SPDX-FileCopyrightText: 2022 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+import Foundation
+import SwiftUI
+
+
+/// A result builder used to aggregate multiple SwiftUI `View`s within the ``ManagedNavigationStack``.
+@resultBuilder
+public enum NavigationFlowBuilder {
```

Comment:
Great idea to just not allow loops here, should save us some trouble.
Maybe a tiny doc, either here or in the managednavigationstack, would be great, indicating that loops are not possible. otherwise, people (especially students), could get confused

Reply:
my worry is that people will try a `for` loop, see it doesn't work, and then simply use a `ForEach` instead. which of course will work, but also won't behave the way we want it to, since it'll all become a single step within the Stack.

we could add `buildX` functions for both `for` loops as well as `ForEach` views, and mark them as unavailable, and then have a message explaining why, what's your take on that @philippzagar?

Reply:
@lukaskollmer I think this is reasonable that we don't cover every single edge case as long as we document it well. Meaning, your solution to detect regular `for` loops but not SwiftUIs `ForEach` is totally fine from my side, thanks for investigating!